openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -6,331 +6,129 @@ package e2e
 
-	}
-}
+		createAppstudioApp(ta)
+		createAppstudioComponent(ta)
 
-func dumpBadEvents(ta *testArgs) {
-	eventClient := kubeClient.EventsV1().Events(ta.ns)
-	eventList, err := eventClient.List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		ta.Logf(fmt.Sprintf("error listing events: %s", err.Error()))
-		return
-	}
-	ta.Logf(fmt.Sprintf("dumpBadEvents have %d items in total list", len(eventList.Items)))
-	for _, event := range eventList.Items {
-		if event.Type == corev1.EventTypeNormal {
-			continue
-		}
-		ta.Logf(fmt.Sprintf("non-normal event reason %s about obj %s:%s message %s", event.Reason, event.Regarding.Kind, event.Regarding.Name, event.Note))
-	}
-}
-
-func dumpNodes(ta *testArgs) {
-	nodeClient := kubeClient.CoreV1().Nodes()
-	nodeList, err := nodeClient.List(context.TODO(), metav1.ListOptions{})
-	if err != nil {
-		ta.Logf(fmt.Sprintf("error listin nodes: %s", err.Error()))
-		return
-	}
-	ta.Logf(fmt.Sprintf("dumpNodes found %d nodes in list, but only logging worker nodes", len(nodeList.Items)))
-	for _, node := range nodeList.Items {
-		_, ok := node.Labels["node-role.kubernetes.io/master"]
-		if ok {
-			continue
-		}
-		if node.Status.Allocatable.Cpu() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have allocatable cpu", node.Name))
-			continue
+		labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"build.appstudio.openshift.io/component": ta.appstudioComponent.Name}}
+		listOptions := metav1.ListOptions{
+			LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+			Limit:         1,
 		}
-		if node.Status.Allocatable.Memory() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have allocatable mem", node.Name))
-			continue
-		}
-		if node.Status.Capacity.Cpu() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have capacity cpu", node.Name))
-			continue
-		}
-		if node.Status.Capacity.Memory() == nil {
-			ta.Logf(fmt.Sprintf("Node %s does not have capacity mem", node.Name))
-			continue
-		}
-		alloccpu := node.Status.Allocatable.Cpu()
-		allocmem := node.Status.Allocatable.Memory()
-		capaccpu := node.Status.Capacity.Cpu()
-		capacmem := node.Status.Capacity.Memory()
-		ta.Logf(fmt.Sprintf("Node %s allocatable CPU %s allocatable mem %s capacity CPU %s capacitymem %s",
-			node.Name,
-			alloccpu.String(),
-			allocmem.String(),
-			capaccpu.String(),
-			capacmem.String()))
-	}
-}
-
-func debugAndFailTest(ta *testArgs, failMsg string) {
-	dumpPods(ta, ta.ns)
-	dumpPods(ta, "jvm-build-service")
-	dumpBadEvents(ta)
-	ta.t.Fatalf(failMsg)
-}
-
-func setup(t *testing.T, ta *testArgs) *testArgs {
-	if ta == nil {
-		ta = &testArgs{
-			t:        t,
-			timeout:  time.Minute * 10,
-			interval: time.Second * 15,
+		err = wait.PollImmediate(ta.interval, ta.timeout, func() (done bool, err error) {
+			prs, err := tektonClient.TektonV1beta1().PipelineRuns(ta.ns).List(context.TODO(), listOptions)
+			if err != nil {
+				return false, nil
```

Comment:
please log the error before returning with ta.Logf
invariably when we have some weird apiservier issue, that log will help